reddit/index.js

```diff
@@ -1,47 +1,113 @@
 const snoowrap = require('snoowrap');
-const { connectToACollection, insertOrUpdateReport } = require('../database');
-
-const {
-  CLIENT_ID: clientId,
-  CLIENT_SECRET: clientSecret,
-  REFRESH_TOKEN: refreshToken,
-  SUBREDDIT: subreddit,
-} = process.env;
-
-const reddit = new snoowrap({
-  userAgent: 'knicksbot',
-  clientId,
-  clientSecret,
-  refreshToken,
-});
-
-/**
- *
- * @param {array} modqueue - Array of items that need to be moderated
- */
-const checkForNewReports = async (modqueue) => {
-  // console.log(JSON.stringify(modqueue));
-  modqueue.forEach(async (report) => {
-    console.log(reddit.getUser(report.author.name));
-    // await insertOrUpdateReport({
-    //   ...report,
-    //   isComment: Boolean(report.body),
-    // });
-  });
-};
-
-/**
- * Starts the scan process of checking for new reported items
- */
-const startReportScan = async () => {
-  return new Promise(async (resolve, reject) => {
-    const modqueue = await reddit.getSubreddit(subreddit).getModqueue();
-    if (modqueue.length > 0) return resolve(await checkForNewReports(modqueue));
-    return reject();
-  });
-};
+const { SubmissionStream, CommentStream, ModQueueStream } = require("snoostorm");
+const { KnicksDiscordBot } = require("../discord");
+const { insertOrUpdateReport } = require("../database")
+
+const KnicksRedditBot = class KnicksRedditBot extends snoowrap {
+ /**
+  * Constructs a new instance.
+  *
+  * @param      {string}  clientId                 The client identifier
+  * @param      {string}  clientSecret             The client secret
+  * @param      {string}  refreshToken             The refresh token
+  * @param      {string}  [userAgent="knicksbot"]  The user agent
+  */
+  constructor (clientId, clientSecret, refreshToken, subreddit, userAgent = "knicksbot") {
+    super({
+      userAgent,
+      clientId,
+      clientSecret,
+      refreshToken
+    });
+
+    this.subreddit = subreddit;
+
+    //Temporary until database set up
+    this._linksList = []
+
+    //Discord bot setup
+    this.discordBot = new KnicksDiscordBot(process.env.BOT_TOKEN);
+    this.discordBot.initialise(process.env.BOT_TOKEN);
+    this.discordBot.commandCollection.on(
+      "ran", async (ctx, args, res, command) => {
+        await this.handleDiscordCommands(ctx, args, res, command);
+      });
+
+    //Snoostorm streams setup
+    this.modQueueStream = new ModQueueStream(this, {
+      subreddit,
+      limit: 50,
+      pollTime: 2000
+    });
+    this.modQueueStream.on("item", async (submission) => {
+      await this.checkSubmission(submission);
+    });
+
```

Comment:
I'd move a lot of these out of the constructor. Despite the fact that they are needed on instantiation, it should be yanked out into an "init" method that is in the class. The constructor should just "set up" the class, but something else should make it usable

Reply:
Moved stream setup and discord bot setup to their own functions